system_setup/server/controllers/wslconfadvanced.py

```diff
@@ -0,0 +1,190 @@
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+
+import attr
+from os import path
+import configparser
+from subiquitycore.context import with_context
+
+from subiquity.common.apidef import API
+from subiquity.common.types import WSLConfigurationAdvanced
+from subiquity.server.controller import SubiquityController
+
+log = logging.getLogger(
+    'subiquity.server.controllers.wsl_configuration_advanced')
+
+
+# TODO WSL: remove all duplicates from WSL config base controller
+class WSLConfigurationAdvancedController(SubiquityController):
+
+    endpoint = API.wslconfadvanced
+
+    autoinstall_key = model_name = "wslconfadvanced"
+    autoinstall_schema = {
+        'type': 'object',
+        'properties': {
+            'custom_path': {'type': 'string'},
+            'custom_mount_opt': {'type': 'string'},
+            'gen_host': {'type': 'boolean'},
+            'gen_resolvconf': {'type': 'boolean'},
+            'interop_enabled': {'type': 'boolean'},
+            'interop_appendwindowspath': {'type': 'boolean'},
+            'gui_theme': {'type': 'string'},
+            'gui_followwintheme': {'type': 'boolean'},
+            'legacy_gui': {'type': 'boolean'},
+            'legacy_audio': {'type': 'boolean'},
+            'adv_ip_detect': {'type': 'boolean'},
+            'wsl_motd_news': {'type': 'boolean'},
+            'automount': {'type': 'boolean'},
+            'mountfstab': {'type': 'boolean'}
+        },
+        'required': [],
+        'additionalProperties': False,
+    }
+
+    # this is a temporary simplified reference. The future complete reference
+    # should use the default.json in `ubuntu-wsl-integration`.
+    config_ref = {
+        "wsl": {
+            "automount": {
+                "enabled": "automount",
+                "mountfstab": "mountfstab",
+                "root": "custom_path",
+                "options": "custom_mount_opt",
+            },
+            "network": {
+                "generatehosts": "gen_host",
+                "generateresolvconf": "gen_resolvconf",
+            },
+            "interop": {
+                "enabled": "interop_enabled",
+                "appendwindowspath": "interop_appendwindowspath",
+            }
+        },
+        "ubuntu": {
+            "GUI": {
+                "theme": "gui_theme",
+                "followwintheme": "gui_followwintheme",
+            },
+            "Interop": {
+                "guiintegration": "legacy_gui",
+                "audiointegration": "legacy_audio",
+                "advancedipdetection": "adv_ip_detect",
+            },
+            "Motd": {
+                "wslnewsenabled": "wsl_motd_news",
+            }
+        }
+    }
+
+    def __init__(self, app):
+        super().__init__(app)
+
+        # load the config file
+        data = {}
+        if path.exists('/etc/wsl.conf'):
```

Comment:
do we want to substitute a placeholder config file when in dry run?

Reply:
This is part of the request to read and write in .subiquity in dryrun mode and part of the TODO list of fixes after this PR is merged.